api/src/main/java/org/infinispan/api/Infinispan.java

```diff
@@ -1,50 +1,99 @@
 package org.infinispan.api;
 
-import java.util.concurrent.CompletionStage;
+import java.net.URI;
+import java.util.ServiceLoader;
 
-import org.infinispan.api.configuration.ClientConfig;
-import org.infinispan.api.configuration.EmbeddedConfig;
-import org.infinispan.api.reactive.KeyValueStore;
-import org.infinispan.api.reactive.KeyValueStoreConfig;
+import org.infinispan.api.async.AsyncContainer;
+import org.infinispan.api.configuration.Configuration;
+import org.infinispan.api.exception.InfinispanConfigurationException;
+import org.infinispan.api.mutiny.MutinyContainer;
+import org.infinispan.api.sync.SyncContainer;
 
 /**
  * Infinispan instance, embedded or client, depending on the access point.
  *
- * @author Katia Aresti, karesti@redhat.com
- * @since 10.0
+ * @since 14.0
  */
 @Experimental("This is not ready yet for general consumption. Major changes are still expected.")
-public interface Infinispan {
-
-   static Infinispan newClient(ClientConfig config) {
-      try {
-         return (Infinispan) Infinispan.class.getClassLoader()
-               .loadClass("org.infinispan.api.client.impl.InfinispanClientImpl")
-               .getConstructor(ClientConfig.class)
-               .newInstance(config);
-      } catch (Exception e) {
+public interface Infinispan extends AutoCloseable {
+   /**
+    * Creates and starts the Infinispan manager object
+    *
+    * Calling create with the same URI multiple times should return the same object. Ref count the object.
```

Comment:
Not sure what `Ref count the object.` here means. Does the user have to do something? Maybe the documentation should be something like as long as the returned manager is still open additional invocations of the same URI will return the same instance without starting anything new. The ref counting is an impl detail imo.